### PR TITLE
feat: add tda and consent config node

### PR DESCRIPTION
**Pull request by cowan-macady**
implement ENG-3351

- add consent config node
- add trusted data access
- update dependencies
- update protobuf libraries
- update go version
- integration tests to follow

### CODE REVIEW

File: tda/data_access_test.go [hunk 1]
```diff
@@ -0,0 +1,507 @@
...
+				gomock.Eq(&mockGrantConsentRequest),
+				gomock.Any(),
+			).Return(nil, status.Error(mockErrorCode, mockErrorMessage))
+
+			resp, err := client.GrantConsent(context.Background(), &mockGrantConsentRequest)
+			Ω(resp).To(BeNil())
+			Ω(err).To(HaveOccurred())
+			Ω(err).To(test.MatchStatusError(mockErrorCode, mockErrorMessage))
+		})
+	})
```
Comment (Septimus4) [inline tda/data_access_test.go:150]: I would be easier to maintain and less repetitive if you use a DescribeTable.
You can do one DescribeTable by endpoints, and add a entry for each case.

**Review by Septimus4 — CHANGES_REQUESTED** [review]

File: tda/data_access_test.go (hunk 1 above)
Comment (cowan-macady) [reply tda/data_access_test.go:150]: Could you be more specific because I think there is a describe table by endpoint?

Comment (Septimus4) [reply tda/data_access_test.go:150]: Ginko support DescribeTable tests
https://github.com/indykite/jarvis-proto/blob/40aec0cb656637fae24af692f6c5f7469bc21d1e/pkg/flow/auth/create_credentials_node_test.go#L123-L122

Comment (cowan-macady) [reply tda/data_access_test.go:150]: ah yes!

**Review by Septimus4 — APPROVED** [review]

**Pull request merged**